app/models/concerns/mentionable.rb

```diff
@@ -0,0 +1,42 @@
+module Mentionable
+  extend ActiveSupport::Concern
+
+  def set_mentioned_users
+    user_ids = extract_user_ids_from_message_body
+    return if user_ids.nil?
+
+    format_message_body
+    add_mentions_lookup(user_ids)
+  end
+
+  def add_mentions_lookup(user_ids)
+    lookup = {}
+    lookup_ids = parse_id_from_decorated_ids(parse_id_from_mention_text)
+    lookup_ids.map.with_index do |lookup_id, idx|
+      lookup[lookup_id] = { type: 'User', id: user_ids[idx], display_name: User.find(user_ids[idx]).full_name }
```

Comment:
Can we cache `display_name`? 🤔   
```
def display_name(user_ids, idx):
    user ||= User.find(user_ids[idx]).full_name
end
```

Reply:
that would be problematic, if in case there are multiple users mentioned, it would always display the cached user name

Reply:
Yeah.. make sense. I thought different id will fetch new result.